schemas/v3.1/meta/base.schema.json

```
@@ -18,6 +18,16 @@
         "externalDocs": { "$ref": "#/$defs/external-docs" },
         "xml": { "$ref": "#/$defs/xml" }
     },
+    "if": {
+      "required": [ "discriminator" ]
+    },
+    "then": {
+      "anyOf": [
+        { "required": [ "oneOf" ] },
+        { "required": [ "anyOf" ] },
+        { "required": [ "allOf" ] }
+      ]
+    },
```

I suggest going with `dependentSchemas` for this. The end result is the same, but it's a little more concise.

```suggestion
    "dependentSchemas": {
      "discriminator": {
        "anyOf": [
          { "required": [ "oneOf" ] },
          { "required": [ "anyOf" ] },
          { "required": [ "allOf" ] }
        ]
      }
    },
```

---

> I wish the spec said that `discriminator` should be ignored if there is no `anyOf`/`oneOf`/`allOf`. That would be more inline with JSON Schema conventions, but that's not what it says, so this change should be included.

Indeed. This would be worthy of reconsidering for v3.2.

> I suggest going with `dependentSchemas` for this. The end result is the same, but it's a little more concise.

Heh, I considered `dependentRequired` but totally forgot about `dependentSchemas`! :p